core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java

```diff
@@ -138,7 +139,7 @@ public JavaScriptTransform(Config config) {
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
     super.configurePipeline(pipelineConfigurer);
     // try evaluating the script to fail application creation if the script is invalid
-    init(null);
+    init(null, pipelineConfigurer.getStageConfigurer().getFailureCollector());
 
     // init intializes schema if present in the config
     Schema outputSchema = (schema == null) ? pipelineConfigurer.getStageConfigurer().getInputSchema() : schema;
```

Comment:
Not your change, but change schema == null check to Strings.isNullOrEmpty(schema). We should be checking for both null and empty.

Reply:
Would we be able to do this? schema is an object and not a string.

Reply:
right, this only applies to schema string.